### PR TITLE
3/create member screen

### PR DESCRIPTION
**Pull request by HadiKhai**
Hello @pyphilia !

This pull request solves #3

### CODE REVIEW

File: src/components/members/MemberScreen.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import itemData from '../../data/itemData';
+import ItemsTable from '../items/ItemsTable';
+
+const MemberScreen = () => {
+  const match = useRouteMatch(buildMemberPath());
+  const memberId = match?.params?.memberId;
+  const member = membersData.find(({ id }) => id === memberId);
+  const itemsWithCreators = insertCreatorWithItems(itemData);
+
+  const memberships = getMembershipsByMemberId(member?.id);
```
Comment (pyphilia) [inline src/components/members/MemberScreen.js:20]: Can you use `memberId`?

File: src/components/members/MemberScreen.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  const itemsPaths = memberships.map(({ itemPath }) => itemPath);
+
+  const items = itemsWithCreators.filter(({ path }) =>
+    itemsPaths.includes(path),
+  );
+
+  return (
+    <div style={{ width: '100%' }}>
```
Comment (pyphilia) [inline src/components/members/MemberScreen.js:29]: Is this necessary?

File: src/components/members/MemberScreen.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  const match = useRouteMatch(buildMemberPath());
+  const memberId = match?.params?.memberId;
+  const member = membersData.find(({ id }) => id === memberId);
+  const itemsWithCreators = insertCreatorWithItems(itemData);
+
+  const memberships = getMembershipsByMemberId(member?.id);
+
+  const itemsPaths = memberships.map(({ itemPath }) => itemPath);
+
+  const items = itemsWithCreators.filter(({ path }) =>
```
Comment (pyphilia) [inline src/components/members/MemberScreen.js:24]: It's `members` right?

File: src/components/members/MemberScreen.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        bgcolor="background.paper"
+      >
+        <Box
+          display="flex"
+          flexDirection="column"
+          p={1}
+          m={1}
+          bgcolor="background.paper"
+        >
+          <AccountCircleIcon style={{ fontSize: 150, width: '100%' }} />
```
Comment (pyphilia) [inline src/components/members/MemberScreen.js:44]: Factor out.

File: src/components/members/MemberScreen.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          <Typography>{`Name: ${member?.name}`}</Typography>
+          <Typography>{`Email: ${member?.email}`}</Typography>
+          <Typography>{`Type: ${member?.type}`}</Typography>
+          <Typography>
+            {`Created At: ${formatDate(member?.createdAt)}`}
+          </Typography>
+        </Box>
+      </Box>
+      <ItemsTable
+        id={`Member_${memberId}_Item_Table`}
```
Comment (pyphilia) [inline src/components/members/MemberScreen.js:55]: Factor out?

File: src/components/members/MemberScreen.js (hunk 3 above)
Comment (pyphilia) [reply src/components/members/MemberScreen.js:24]: But I guess these will be changes once you will connect the db right?

File: src/components/members/MembersTable.js [hunk 6]
```diff
@@ -71,23 +67,34 @@ const MembersTable = ({ members: rows, tableTitle, id: tableId, empty }) => {
...
+  const [searchValue, setSearchValue] = React.useState('');
+
   useEffect(() => {
-    // remove deleted rows from selection
-    const newSelected = selected.filter(
-      (id) => rows.findIndex(({ id: thisId }) => thisId === id) >= 0,
-    );
-    if (newSelected.length !== selected.length) {
-      setSelected(newSelected);
+    if (searchValue === '') {
```
Comment (pyphilia) [inline src/components/members/MembersTable.js:88]: Check length.

File: src/components/members/MembersTable.js [hunk 7]
```diff
@@ -271,7 +325,7 @@ const MembersTable = ({ members: rows, tableTitle, id: tableId, empty }) => {
         <TablePagination
           rowsPerPageOptions={ROWS_PER_PAGE_OPTIONS}
           component="div"
-          count={rows.size}
+          count={filteredRows.size}
           rowsPerPage={rowsPerPage}
           page={page}
           onChangePage={handleChangePage}
```
Comment (pyphilia) [inline src/components/members/MembersTable.js:331]: Looks like a lot like MembersTable. Is it possible to factor out similar components?

File: src/data/membersData.js [hunk 8]
```diff
@@ -28,8 +28,8 @@ const membersData = [
   },
   {
     id: 'd2c4875f-e721-4cba-be90-e0dc1538181a',
-    name: 'Hadi 2',
-    email: 'hadikhairallah@outlook.com',
+    name: 'Denis ',
+    email: 'denis.gillet@epfl.ch',
```
Comment (pyphilia) [inline src/data/membersData.js:32]: Actually could you avoid using real data? Like anna@email.com. Names are okay, but mail a bit less.

**Review by pyphilia — CHANGES_REQUESTED** [review]
Nice page, I left some comments let me know if you have questions.
I also noticed this weird annotation next to the column you select to sort (`sorted descending`)
Screenshot 2021-07-12 at 10 30 30

File: src/components/members/MemberScreen.js (hunk 1 above)
Comment (HadiKhai) [reply src/components/members/MemberScreen.js:20]: Everything related to the mock data has been removed in #12

File: src/components/members/MembersTable.js (hunk 7 above)
Comment (HadiKhai) [reply src/components/members/MembersTable.js:331]: I will open an issue to factoring out all the tables

File: src/components/items/ItemsTable.js [hunk 9]
```diff
@@ -58,6 +58,10 @@ const useStyles = makeStyles((theme) => ({
   itemName: {
     paddingLeft: theme.spacing(1),
   },
+  autoComplete: {
+    width: 300,
```
Comment (pyphilia) [inline src/components/items/ItemsTable.js:62]: Factor out ;)

**Review by pyphilia — APPROVED** [review]
Good! I left minor comments, you can merge after resolving them 👍

**Pull request merged**